### PR TITLE
Dns policy ingress for RKE

### PR DESCRIPTION
**Pull request by Bourne-ID**
Adds the dnsPolicy to the RKE Ingress section when creating a cluster. Documentation updated.

### CODE REVIEW

File: website/docs/r/cluster.html.markdown [hunk 1]
```diff
@@ -450,6 +450,7 @@ The following attributes are exported:
 
 ##### Arguments
 
+* `dns_policy` - (Optional/Computed) DNS Policy for RKE Ingress (map)
```
Comment (rawmind0) [inline website/docs/r/cluster.html.markdown]: Wrong argument type. Also adding supported values.

```
- * `dns_policy` - (Optional/Computed) DNS Policy for RKE Ingress (map)
+ * `dns_policy` - (Optional/Computed) Ingress controller DNS policy. `ClusterFirstWithHostNet`, `ClusterFirst`, `Default`, and `None` are supported. [K8S dns Policy](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy) (string)
```

File: rancher2/schema_cluster_rke_config_ingress.go [hunk 2]
```diff
@@ -28,6 +28,11 @@ func clusterRKEConfigIngressFields() map[string]*schema.Schema {
 			Optional: true,
 			Computed: true,
 		},
+		"dns_policy": {
+			Type:     schema.TypeString,
+			Optional: true,
+			Computed: true,
+		},
```
Comment (rawmind0) [inline rancher2/schema_cluster_rke_config_ingress.go:53]: What do you think to add argument validation to limit the value of `dns_policy` just to available at [K8S dns Policy](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-dns-policy) ??

```
import (
	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
)

+const (
+	clusterRKEConfigIngressDNSPolicyClusterFirst = "ClusterFirst"
+	clusterRKEConfigIngressDNSPolicyClusterFirstWithHostNet = "ClusterFirstWithHostNet"
+	clusterRKEConfigIngressDNSPolicyDefault = "Default"
+	clusterRKEConfigIngressDNSPolicyNone = "None"
+)
+
+var (
+	clusterRKEConfigIngressDNSPolicyList = []string{
+		clusterRKEConfigIngressDNSPolicyClusterFirst, 
+		clusterRKEConfigIngressDNSPolicyClusterFirstWithHostNet,
+		clusterRKEConfigIngressDNSPolicyDefault,
+		clusterRKEConfigIngressDNSPolicyNone
+	}
+)
...
		"dns_policy": {
			Type:     schema.TypeString,
			Optional: true,
			Computed: true,
+			ValidateFunc: validation.StringInSlice(clusterRKEConfigIngressDNSPolicyList, true),
		},
```

**Review by rawmind0 — CHANGES_REQUESTED** [review]
@Bourne-ID , thanks for the PR. 

Could you please take a look to the changes requested??

File: rancher2/schema_cluster_rke_config_ingress.go (hunk 2 above)
Comment (Bourne-ID) [reply rancher2/schema_cluster_rke_config_ingress.go:53]: I thought about this but hesitated as tests on the Rancher API shows it accepts any string (even if this means the cluster fails as the dns policy is not recognised) - is it correct to add restrictions on the Terraform provider when there are no restrictions on Rancher? I'm for this change, but at the same time don't want to cause a limitation if there has been a decision to make this an any-entry string on Rancher...

File: website/docs/r/cluster.html.markdown (hunk 1 above)
Comment (Bourne-ID) [reply website/docs/r/cluster.html.markdown]: Great catch - fixed and agreed with supported values + link.

File: rancher2/schema_cluster_rke_config_ingress.go (hunk 2 above)
Comment (Oats87) [reply rancher2/schema_cluster_rke_config_ingress.go:53]: The Rancher API accepts anything but RKE itself should not: https://github.com/rancher/rke/blob/master/cluster/validation.go#L171

Comment (Bourne-ID) [reply rancher2/schema_cluster_rke_config_ingress.go:53]: Changes added

**Review by rawmind0 — APPROVED** [review]
LGTM

**Pull request merged**